lib/internal/bootstrap/node.js

```diff
@@ -475,6 +475,7 @@
         try {
           if (!process._exiting) {
             process._exiting = true;
+            process.exitCode = 1;
```

Comment:
If user code happens to set `process.exitCode` prior to this, this will override the user provided value. It should likely only set the value if it is not already set

Reply:
I do believe this is not the case, as uncaughtException has its own code 1, so it should be correct to override whatever was the code before. Though this is probably a semver-major because of this, but that shouldn't be a problem imo.

Reply:
hmm... I can definitely see the logic on it but I definitely don't like overriding user provided values unexpectedly. If we go with this, can I ask that a note be added to the documentation along with a code comment indicating why it's ok to silently override any user provided value here?

Reply:
That's fine with me. But where should I put a documentation?
 Also, it is already [noted](https://nodejs.org/dist/latest-v10.x/docs/api/process.html#process_exit_codes) that unhandledException has code 1, so this will basically enforce this, so that any previously set code will not be used, only the ones set in 'unhandledException' callback, 'exit' callback etc.

Reply:
> But where should I put a documentation?

In the description for the `uncaughtException` event in `docs/api/process.md`. The documentation currently does not say anything about the process exit code. A note there that, should the event *not* be handled, the process will exit with exitCode = 1 *even if* the user had previously set a `process.exitCode` value.

Reply:
(btw, I see this as a limitation in the *current* documentation and not something that is introduced by this PR)

Reply:
> even if the user had previously set a process.exitCode value.

Oh, I thought that's it's a given that if some new error happen exitCode will be replaced with the value of the error and just thought that it was undocumented. The fact that you wasn't able to change the code is indeed a strange thing. Anyway, I'll add the doc soon.